### PR TITLE
Added technical debts uploading from rules definitions file

### PR DESCRIPTION
**Pull request by leszekszczepanski**
Added technical debts uploading from rules definitions file

### CODE REVIEW

**Comment by julienlancelot** [comment]
Hi,

Thanks for your contribution.

Here are my first 2 feedbacks : 
- Please use our coding convention : http://docs.sonarqube.org/display/SONAR/Hosting+on+the+Forge (See last part "IDE Support"), otherwise it's impossible to review
- Add Unit Tests

Thanks

**Comment by leszekszczepanski** [comment]
Hi,

Corrected.
Is it possible to add here also merge to 5.2 or should I do that once again for branch-5.2?

Thanks

File: sonar-plugin-api/src/test/java/org/sonar/api/server/rule/RulesDefinitionXmlLoaderTest.java [hunk 1]
```diff
@@ -57,6 +57,9 @@ public void parse_xml() throws Exception {
     assertThat(rule.severity()).isEqualTo(Severity.BLOCKER);
     assertThat(rule.template()).isTrue();
     assertThat(rule.status()).isEqualTo(RuleStatus.BETA);
+    assertThat(rule.debtSubCharacteristic()).isEqualTo("INTEGRATION_TESTABILITY");
+    assertThat(rule.debtRemediationFunction().coefficient()).isEqualTo("2d 2h 2min".replaceAll(" ", ""));
```
Comment (julienlancelot) [inline sonar-plugin-api/src/test/java/org/sonar/api/server/rule/RulesDefinitionXmlLoaderTest.java]: Please remove this useless .replaceAll(" ", ""), and replace expected result by "2d2h2min"

File: sonar-plugin-api/src/test/java/org/sonar/api/server/rule/RulesDefinitionXmlLoaderTest.java [hunk 2]
```diff
@@ -57,6 +57,9 @@ public void parse_xml() throws Exception {
...
+    assertThat(rule.debtRemediationFunction().offset()).isEqualTo("1d 1h 1min".replaceAll(" ", ""));
```
Comment (julienlancelot) [inline sonar-plugin-api/src/test/java/org/sonar/api/server/rule/RulesDefinitionXmlLoaderTest.java]: Same as before

Comment (leszekszczepanski) [inline sonar-plugin-api/src/test/java/org/sonar/api/server/rule/RulesDefinitionXmlLoaderTest.java]: Corrected.
I decided to add that .replaceAll(" ", "") as notification that parser during rules uploading is able to validate input like "2d 2h 2min" but value is returned without blanks.

**Comment by julienlancelot** [comment]
Thanks for this contribution, but sorry we won't apply your P/R as we want to have only one way to declare Rules, and the XML API will soon be deprecated then removed.

**Pull request closed without merging**